pkg/component/controller/cplb/cplb_linux.go

```diff
@@ -321,7 +344,68 @@ func (k *Keepalived) generateKeepalivedTemplate() error {
 	return nil
 }
 
-func (k *Keepalived) watchReconcilerUpdates() {
+func (k *Keepalived) watchReconcilerUpdatesReverseProxy() error {
+	k.proxy = tcpproxy.Proxy{}
+	// We don't know how long until we get the first update, so initially we
+	// forward everything to localhost
+	k.proxy.AddRoute(fmt.Sprintf(":%d", k.Config.UserSpaceProxyPort), tcpproxy.To(fmt.Sprintf("127.0.0.1:%d", k.APIPort)))
+
+	if err := k.proxy.Start(); err != nil {
+		return fmt.Errorf("failed to start proxy: %w", err)
+	}
+
+	fmt.Println("Waiting for updateCh")
+	<-k.updateCh
+	k.setProxyRoutes()
+
+	// Do not create the iptables rules until we have the first update and the
+	// proxy is running
+	if err := k.redirectToProxyIPTables(iptablesCommandAppend); err != nil {
+		k.log.Fatal(err)
+	}
+
+	for range k.updateCh {
+		k.setProxyRoutes()
+	}
+	return nil
+}
+
+func (k *Keepalived) setProxyRoutes() {
+	routes := []tcpproxy.Target{}
+	for _, addr := range k.reconciler.GetIPs() {
+		routes = append(routes, tcpproxy.To(fmt.Sprintf("%s:%d", addr, k.APIPort)))
+	}
+
+	k.proxy.SetRoutes(fmt.Sprintf(":%d", k.Config.UserSpaceProxyPort), routes)
+}
+
+func (k *Keepalived) redirectToProxyIPTables(op string) error {
+	for _, vrrp := range k.Config.VRRPInstances {
+		for _, vipCIDR := range vrrp.VirtualIPs {
+			vip := strings.Split(vipCIDR, "/")[0]
+
+			cmdArgs := []string{
+				"-t", "nat", op, "PREROUTING", "-p", "tcp",
+				"-d", vip, "--dport", strconv.Itoa(k.APIPort),
+				"-j", "REDIRECT", "--to-port", strconv.Itoa(k.Config.UserSpaceProxyPort),
+			}
```

Comment:
The clean up is done`Keepalived.Stop()` method.

Regarding dynamicConfig, cplb isn't managed with dynamiConfig and it's stated in the docs. At least until we have config v2 there is no way for us to make it compatible because the cplb configuration might differ per control plane node.